### PR TITLE
Add Github Issue backend

### PR DESCRIPTION
**Pull request by i386**
# Description

Adds a `BlueIssue` implementation for Github issues and PRs.

See [JENKINS-XXXXX](https://issues.jenkins-ci.org/browse/JENKINS-XXXXX).

# Submitter checklist
- [ ] Link to JIRA ticket in description, if appropriate.
- [ ] Change is code complete and matches issue description
- [ ] Appropriate unit or acceptance tests or explanation to why this change has no tests
- [ ] Reviewer's manual test instructions provided in PR description. See Reviewer's first task below.

# Reviewer checklist
- [ ] Run the changes and verified the change matches the issue description
- [ ] Reviewed the code
- [ ] Verified that the appropriate tests have been written or valid explanation given

### CODE REVIEW

File: blueocean-github-pipeline/src/main/java/io/jenkins/blueocean/blueocean_github_pipeline/GithubIssue.java [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+                return null;
+            }
+            GitHubSCMSource source = (GitHubSCMSource) sources.get(0);
+            final String repositoryUri = source.getUriResolver().getRepositoryUri(source.getApiUri(), source.getRepoOwner(), source.getRepository());
+            return Collections2.transform(findIssueKeys(changeSetEntry.getMsg()), new Function<String, BlueIssue>() {
+                @Override
+                public BlueIssue apply(String input) {
+                    // Remove ".git"
+                    String uri = repositoryUri.substring(0, repositoryUri.length() - 4);
+                    return new GithubIssue("#" + input, String.format("%s/issues/%s", uri, input));
```
Comment (i386) [inline blueocean-github-pipeline/src/main/java/io/jenkins/blueocean/blueocean_github_pipeline/GithubIssue.java]: Note that if the issue is really a PR, Github will redirect to the PR.

**Comment by i386** [comment]
PTAL @vivek

File: blueocean-github-pipeline/src/main/java/io/jenkins/blueocean/blueocean_github_pipeline/GithubIssue.java [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+        @Override
+        public Collection<BlueIssue> getIssues(ChangeLogSet.Entry changeSetEntry) {
+            Job job = changeSetEntry.getParent().getRun().getParent();
+            if (!(job.getParent() instanceof MultiBranchProject)) {
+                return null;
+            }
+            MultiBranchProject mbp = (MultiBranchProject)job.getParent();
+            List sources = mbp.getSCMSources();
+            if (sources.isEmpty() || !(sources.get(0) instanceof GitHubSCMSource)) {
```
Comment (kzantow) [inline blueocean-github-pipeline/src/main/java/io/jenkins/blueocean/blueocean_github_pipeline/GithubIssue.java]: Any reason not to check _all_ sources?

File: blueocean-github-pipeline/src/main/java/io/jenkins/blueocean/blueocean_github_pipeline/GithubIssue.java [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+                return null;
+            }
+            GitHubSCMSource source = (GitHubSCMSource) sources.get(0);
+            final String repositoryUri = source.getUriResolver().getRepositoryUri(source.getApiUri(), source.getRepoOwner(), source.getRepository());
+            return Collections2.transform(findIssueKeys(changeSetEntry.getMsg()), new Function<String, BlueIssue>() {
```
Comment (kzantow) [inline blueocean-github-pipeline/src/main/java/io/jenkins/blueocean/blueocean_github_pipeline/GithubIssue.java]: Aren't we using Java 8 now? There are streams APIs for this type of stuff...

Comment (i386) [inline blueocean-github-pipeline/src/main/java/io/jenkins/blueocean/blueocean_github_pipeline/GithubIssue.java]: I don't believe we are

File: blueocean-github-pipeline/src/main/java/io/jenkins/blueocean/blueocean_github_pipeline/GithubIssue.java (hunk 2 above)
Comment (i386) [inline blueocean-github-pipeline/src/main/java/io/jenkins/blueocean/blueocean_github_pipeline/GithubIssue.java]: By convention elsewhere in the code base we only care about the first SCM source or SCM navigator.

File: blueocean-github-pipeline/src/main/java/io/jenkins/blueocean/blueocean_github_pipeline/GithubIssue.java (hunk 3 above)
Comment (i386) [inline blueocean-github-pipeline/src/main/java/io/jenkins/blueocean/blueocean_github_pipeline/GithubIssue.java]: I checked - looks like we aren't

Comment (i386) [inline blueocean-github-pipeline/src/main/java/io/jenkins/blueocean/blueocean_github_pipeline/GithubIssue.java]: Ended up moving to an guava search that should convert fine to java8 with an IntelliJ refactor

**Pull request merged**